### PR TITLE
Handling both http and https repository path

### PR DESCRIPTION
**Pull request by PavamanSubramaniyam**
Added the support for fetching the MOFED ISO from both http and https repository

### CODE REVIEW

**Comment by PavamanSubramaniyam** [comment]
I have attached the detail execution and complete logs in the corresponding Jira task.

**Comment by PavamanSubramaniyam** [comment]
Below is the result on RHEL9.2 OS with https repository of MOFED ISO.

avocado run --max-parallel-tasks=1 mofed_install_test.py -m mofed_install_test.py.data/mofed_install_test.yaml
Fetching asset from mofed_install_test.py:MOFEDInstallTest.test
Fetching asset from mofed_install_test.py:MOFEDInstallTest.test
JOB ID     : 1b29085844f04bfae6ccac53fb135fbcf50ec09f
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2024-07-22T00.20-1b29085/job.log
 (1/1) mofed_install_test.py:MOFEDInstallTest.test;run-Options-add-kernel-support-Parameters-e70e: STARTED
 (1/1) mofed_install_test.py:MOFEDInstallTest.test;run-Options-add-kernel-support-Parameters-e70e: PASS (807.06 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2024-07-22T00.20-1b29085/results.html
JOB TIME   : 837.68 savocado run --max-parallel-tasks=1 mofed_install_test.py -m mofed_install_test.py.data/mofed_install_test.yaml
Fetching asset from mofed_install_test.py:MOFEDInstallTest.test
Fetching asset from mofed_install_test.py:MOFEDInstallTest.test
JOB ID     : 1b29085844f04bfae6ccac53fb135fbcf50ec09f
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2024-07-22T00.20-1b29085/job.log
 (1/1) mofed_install_test.py:MOFEDInstallTest.test;run-Options-add-kernel-support-Parameters-e70e: STARTED
 (1/1) mofed_install_test.py:MOFEDInstallTest.test;run-Options-add-kernel-support-Parameters-e70e: PASS (807.06 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2024-07-22T00.20-1b29085/results.html
JOB TIME   : 837.68 s

File: io/net/infiniband/mofed_install_test.py [hunk 1]
```diff
@@ -72,10 +77,10 @@ def setUp(self):
                     detected_distro.name, detected_distro.version, detected_distro.release)
                 patterns = [host_distro_pattern]
                 for pattern in patterns:
-                    temp_string = process.getoutput(
-                        "curl --silent %s" % (self.iso_location), verbose=False, shell=True, ignore_status=True)
+                    response = urllib.request.urlopen(self.iso_location)
```
Comment (PraveenPenguin) [inline io/net/infiniband/mofed_install_test.py:80]: can we try to see if `asset_fetch` module from avocado work?

Comment (PavamanSubramaniyam) [reply io/net/infiniband/mofed_install_test.py:80]: We are matching the MOFED ISO to be installed based on the distro OS already installed in the server. 
So, for this scenario we are getting the ISO from the http or https repository wherein many different ISO's will be available for different distro releases.

We are building the full iso path once we match the OS level with the corresponding MOFED ISO and then download the same to the server to install the same.

We are using fetch_asset() function of avocado after we get the full iso path and pass the same to it as can be seen in lines 87 and line 53 of this script.

Comment (abdhaleegit) [reply io/net/infiniband/mofed_install_test.py:80]: @PraveenPenguin `self.iso = self.fetch_asset(self.iso, expire='10d')` its being used already... but before that this code to handle https path

**Review by abdhaleegit — APPROVED** [review]
LGTM, @PavamanSubramaniyam make sure it works for http and https both links.. I see only one log

**Comment by abdhaleegit** [comment]
Please confirm once if it works ?

**Comment by PavamanSubramaniyam** [comment]
Below is the result on RHEL9.2 OS with http repository of MOFED ISO.

avocado run --max-parallel-tasks=1 mofed_install_test.py -m mofed_install_test.py.data/mofed_install_test.yaml
Fetching asset from mofed_install_test.py:MOFEDInstallTest.test
Fetching asset from mofed_install_test.py:MOFEDInstallTest.test
JOB ID     : 804ad65607b1d424afb4847371e3aac2da4370a9
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2024-07-22T01.22-804ad65/job.log
 (1/1) mofed_install_test.py:MOFEDInstallTest.test;run-Options-add-kernel-support-Parameters-d49d: STARTED
 (1/1) mofed_install_test.py:MOFEDInstallTest.test;run-Options-add-kernel-support-Parameters-d49d: PASS (743.48 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2024-07-22T01.22-804ad65/results.html
JOB TIME   : 765.33 s

**Pull request merged**